src/agentlab/analyze/agent_xray.py

```diff
@@ -581,7 +582,9 @@ def update_chat_messages():
     global info
     agent_info = info.exp_result.steps_info[info.step].agent_info
     chat_messages = agent_info.get("chat_messages", ["No Chat Messages"])
-    messages = []
+    if isinstance(chat_messages, Discussion):
+        return chat_messages.to_markdown()
+    messages = []  # TODO(ThibaultLSDC) remove this at some point
     for i, m in enumerate(chat_messages):
         if isinstance(m, BaseMessage):  # TODO remove once langchain is deprecated
             m = m.content
```

Comment:
The Discussion class deprecates a lot code pieces but I figured it might be safer to keep for a while

Reply:
For backward compatibility? Perhaps we can wrap backward compatible code in some isolated function (no need to do now).

As long as we're at laeast forward compatible :)

Reply:
Haven't thought of that. I feel like it would be a weird usecase to use only browsergym on traces that were made with Agentlab though